src/Components/List/ListItem.tsx

```diff
@@ -0,0 +1,201 @@
+// Libraries
+import React, {
+  forwardRef,
+  MouseEvent,
+  createContext,
+  useContext,
+  cloneElement,
+} from 'react'
+import classnames from 'classnames'
+
+// Contexts
+import {ListContext} from './List'
+
+// Components
+import {ListItemHighlight} from './ListItemHighlight'
+
+// Utils
+import {calculateTextColorFromBackground} from '../../Utils'
+
+// Types
+import {
+  StandardFunctionProps,
+  InfluxColors,
+  ComponentSize,
+  Gradients,
+} from '../../Types'
+
+export interface ListItemSharedProps {
+  /** Whether or not the item should have selected styling */
+  selected?: boolean
+  /** Size of this component */
+  size?: ComponentSize
+}
+
+export interface ListItemContextProps extends ListItemSharedProps {
+  listItemContrastColor?: string
+  listItemBackgroundColor?: InfluxColors | string
+  listItemGradient?: Gradients
+}
+
+type CombinedListItemProps = ListItemSharedProps & StandardFunctionProps
+
+export interface ListItemProps extends CombinedListItemProps {
+  /** Value to be returned via the onClick function */
+  value?: any
+  /** When a dropdown item is clicked, its `value` prop is returned via `onChange` */
+  onClick?: (value?: any) => void
+  /** Controls whether the text contents of this item wrap or not */
+  wrapText?: boolean
+  /** Title attribute */
+  title?: string
+  /** Prevents any interaction with this element, including the onClick function */
+  disabled?: boolean
+  /** Pass in an <a> or <Link> element as an alternative to onClick */
+  linkElement?: JSX.Element
+  /** Colorizes the background of the list item in hover and selected state */
+  backgroundColor?: InfluxColors | string
+  /** Overrides backgroundColor, fills background with gradient */
+  gradient?: Gradients
+}
+
+export type ListItemRef = HTMLDivElement
+
+export const ListItemContext = createContext<ListItemContextProps>({
+  selected: false,
+  size: ComponentSize.Small,
+  listItemBackgroundColor: undefined,
```

Comment:
Is `undefined` necessary here? Seems redundant to define a default prop as "undefined".

Reply:
The context needs some initial value set. I could do `null` instead. I could be wrong but I think it will complain if I don't set a value

Reply:
Ya, defining null would be better than defining undefined.